api/types/types.go

```diff
@@ -185,6 +185,9 @@ type Version struct {
 type Info struct {
 	ID                 string
 	Containers         int
+	ContainersRunning  int
+	ContainersPaused   int
+	ContainersStopped  int
```

Comment:
I'd prefer to be more explicit about which fields we're returning. While I agree that a map is a bit more future-proof, it also gives us the freedom to remove fields too easily. Adding, or removing, states should be a pretty big deal - causing an API change isn't unreasonable. So, in this case I think being strongly typed (if that's the right phrase for it) is probably the right choice.